src/tools/tools/StringTools.fs

```diff
@@ -60,6 +61,16 @@ type StringTComparer<'T when 'T:equality>=
             else
                 (17 * 23 + this._comp.GetHashCode( xstring )) * 23 + xval.GetHashCode()
 
+/// <summary>
+/// Construct a comparer that uses Object.ReferenceEquals to compare object. 
+/// </summary>
+type ReferenceComparer<'T>()=
```

Comment:
why it's in StringTools.fs? seems not specific to string.  Maybe in utils.fs?  Also, should it be internal?